code/game/machinery/kitchen/snackbarmachine.dm

```diff
@@ -1,173 +1,52 @@
-/obj/machinery/snackbar_machine
-	name = "SnackBar Machine"
-	density = 1
-	anchored = 1
-	icon = 'icons/obj/chemical.dmi'
-	icon_state = "mixer0"
-	use_power = 1
-	idle_power_usage = 20
-	var/obj/item/weapon/reagent_containers/glass/beaker = null
-	var/mode = 0
-	var/opened = 0
-	var/useramount = 30 // Last used amount
-
-	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE | FIXED2WORK
-
-	light_color = LIGHT_COLOR_CYAN
-	power_change()
-		..()
-		if(!(stat & (BROKEN|NOPOWER)))
-			set_light(2)
-		else
-			set_light(0)
+/obj/machinery/chem_master/snackbar_machine
+	name = "\improper SnackBar Machine"
+	desc = "An explosion of flavour in every bite"
+	condi = 1
+	chem_board = /obj/item/weapon/circuitboard/snackbar_machine
+	//var/opened = 0
 
-/********************************************************************
-**   Adding Stock Parts to VV so preconstructed shit has its candy **
-********************************************************************/
-/obj/machinery/snackbar_machine/New()
-	. = ..()
-	create_reagents(100)
-
-	component_parts = newlist(\
-		/obj/item/weapon/circuitboard/snackbar_machine,\
-		/obj/item/weapon/stock_parts/manipulator,\
-		/obj/item/weapon/stock_parts/manipulator,\
-		/obj/item/weapon/stock_parts/scanning_module,\
-		/obj/item/weapon/stock_parts/scanning_module,\
-		/obj/item/weapon/stock_parts/micro_laser,\
-		/obj/item/weapon/stock_parts/micro_laser,\
-		/obj/item/weapon/stock_parts/console_screen,\
-		/obj/item/weapon/stock_parts/console_screen\
-	)
-
-	RefreshParts()
-
-	overlays += image('icons/obj/chemical.dmi', src, "[icon_state]_overlay")
-
-/obj/machinery/snackbar_machine/ex_act(severity)
-	switch(severity)
-		if(1.0)
-			qdel(src)
-			return
-		if(2.0)
-			if (prob(50))
-				qdel(src)
-				return
-
-/obj/machinery/snackbar_machine/blob_act()
-	if (prob(50))
-		qdel(src)
-
-/obj/machinery/snackbar_machine/power_change()
-	if(powered())
-		stat &= ~NOPOWER
-	else
-		spawn(rand(0, 15))
-			stat |= NOPOWER
+/obj/machinery/chem_master/snackbar_machine/attackby(var/obj/item/weapon/B as obj, var/mob/user as mob)
 
-/obj/machinery/snackbar_machine/attackby(var/obj/item/weapon/B as obj, var/mob/user as mob)
+	if(istype(B, /obj/item/weapon/storage/pill_bottle))
+		user << "<span class='warning'>This condiment master does not come with a pill dispenser unit built-in.</span>"
+		return 1
 
-	if(istype(B, /obj/item/weapon/reagent_containers/glass))
-		if(src.beaker)
-			user << "A beaker is already loaded into the machine."
-			return
-		src.beaker = B
-		user.drop_item(B, src)
-		user << "You add the beaker to the machine!"
-		src.updateUsrDialog()
-		update_icon()
+	. = ..()
 
-	else
-		..()
+/obj/machinery/chem_master/snackbar_machine/Topic(href, href_list)
 
-/obj/machinery/snackbar_machine/Topic(href, href_list)
-	if(..()) return 1
+	if(..())
+		return 1
 
-	src.add_fingerprint(usr)
 	usr.set_machine(src)
 
 	if(href_list["close"])
```

Comment:
When you called ..(), it gets caught in the parent's "close" and returned 1 canceling the topic before getting here